### PR TITLE
Change the recommendation logic

### LINKED ISSUE #35: [FEAT][IMPL] - Stack recommendation logic

### PR DESCRIPTION
**Pull request by arunkumars08**
auth.py
    1. Add a generic function to decode auth token

api_v1.py
    1. Get the decoded token
    2. Pass it along with the stack analysis call of Graph Aggregator

Fixes #35

### CODE REVIEW

**Comment by msrb** [comment]
https://github.com/fabric8-analytics/fabric8-analytics-server/pull/44 should fix CI.

**Comment by pkajaba** [comment]
[test this]

File: bayesian/api_v1.py [hunk 1]
```diff
@@ -152,6 +152,19 @@ def inner(*args, **kwargs):
 #  pass through add_resource_no_matter_slashes
 _resource_paths = []
 
+# START - TOKEN DECODE
```
Comment (pkajaba) [inline bayesian/api_v1.py]: I don't think that this start end statement is really necessary,

File: bayesian/api_v1.py [hunk 2]
```diff
@@ -152,6 +152,19 @@ def inner(*args, **kwargs):
...
+def get_decoded_token():
+    token = request.headers.get('Authorization')
+    if token.startswith('Bearer '):
+        _, token = token.split(' ', 1)
+    try:
+        decoded_token = decode_token(token)
+    except:
```
Comment (fridex) [inline bayesian/api_v1.py]: You would probably want to catch `ExpiredSignatureError` here as well to report expired token as stated in https://github.com/fabric8-analytics/fabric8-analytics-server/issues/55

File: bayesian/auth.py [hunk 3]
```diff
@@ -15,6 +15,16 @@
 
 jwt.register_algorithm('RS256', RSAAlgorithm(RSAAlgorithm.SHA256))
 
+# START - USER PROFILING
```
Comment (fridex) [inline bayesian/auth.py]: Probably a debug leftover, we don't need `# START` comments here.

File: bayesian/auth.py [hunk 4]
```diff
@@ -15,6 +15,16 @@
...
+def decode_token(token):
```
Comment (fridex) [inline bayesian/auth.py]: This looks like a duplicit logic with `get_decoded_token`.

**Comment by samuzzal-choudhury** [comment]
@arunkumars08 I think that this PR is not valid any more as the corresponding changes have already been done during the Bookkeeper task implementation. Please verify and close the PR if you think so.

**Comment by arunkumars08** [comment]
Closing this PR as it is not applicable anymore.

**Pull request closed without merging**